### PR TITLE
Omit obtaining the lock when adding Nodes in newNodeTree

### PR DESCRIPTION
**Pull request by tedyu**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
In newNodeTree, we take lock, add a Node and release the lock.
This is repeated for every Node to be added.

This PR obtains lock and keeps the lock for all the Nodes added.

```release-note
NONE
```

### CODE REVIEW

File: pkg/scheduler/internal/cache/node_tree.go [hunk 1]
```diff
@@ -62,8 +62,13 @@ func newNodeTree(nodes []*v1.Node) *NodeTree {
 	nt := &NodeTree{
 		tree: make(map[string]*nodeArray),
 	}
+	if len(nodes) == 0 {
+		return nt
+	}
+	nt.mu.Lock()
+	defer nt.mu.Unlock()
```
Comment (ahg-g) [inline pkg/scheduler/internal/cache/node_tree.go]: You don't need to acquire the lock at all, at this point nt is still a local variable.

File: pkg/scheduler/internal/cache/node_tree.go [hunk 2]
```diff
@@ -62,8 +62,13 @@ func newNodeTree(nodes []*v1.Node) *NodeTree {
 	nt := &NodeTree{
 		tree: make(map[string]*nodeArray),
 	}
+	if len(nodes) == 0 {
+		return nt
+	}
```
Comment (ahg-g) [inline pkg/scheduler/internal/cache/node_tree.go]: we don't need this check, the code below is generic enough to return an empty nt if nodes is empty.

**Comment by ahg-g** [comment]
can you please remove the bug label? I don't think I can remove it, and this is not a bug.

**Comment by tedyu** [comment]
@ahg-g
Can you take another look ?

**Comment by tedyu** [comment]
@wojtek-t
Can you approve this ?

**Review by Huang-Wei — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**